pkg/hubble/server/serveroption/option.go

```diff
@@ -34,7 +34,7 @@ import (
 // Options stores all the configuration values for the hubble server.
 type Options struct {
 	TCPListener        net.Listener
-	UnixSocketListener net.Listener
+	UnixSocketListener *net.UnixListener
```

Comment:
PR #12906 (still draft) will remove support for multiple listeners as this is a feature that is unused. However, this means that `WithUnixSocketListener()` and `WithTCPListener()`, which are mutually exclusive options, both set a single listener which has to be of type `net.Listener`. Can we somehow keep `net.Listener` as the type here? I'll have to revert that change anyway when rebasing #12906 against master with the changes in this PR.

Reply:
@Rolinh we'll block this PR waiting on yours as this one is simpler and less impactful. If we end up with a single listener that is not necessarily a socket, then it has to be a `net.Listener` of course.

Reply:
Kept `net.Listener` as discussed when rebased.